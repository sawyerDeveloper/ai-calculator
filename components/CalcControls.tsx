import { StyleSheet, View } from 'react-native';
import { CalcButton } from './CalcButton';

interface CalcControlsProps {
  pressCalcButton: (calcValue: number | string) => void;
}

export default function CalcControls({ pressCalcButton }: CalcControlsProps) {
  return (
    <View style={styles.container}>
      <View style={styles.column}>
        <View style={styles.row}>
          <CalcButton label='1' value={1} action={pressCalcButton} />
          <CalcButton label='2' value={2} action={pressCalcButton} />
          <CalcButton label='3' value={3} action={pressCalcButton} />
          <CalcButton
            label='+'
            value={'+'}
            action={pressCalcButton}
            backgroundColor='lightgrey'
          />
        </View>
        <View style={styles.row}>
          <CalcButton label='4' value={4} action={pressCalcButton} />
          <CalcButton label='5' value={5} action={pressCalcButton} />
          <CalcButton label='6' value={6} action={pressCalcButton} />
          <CalcButton
            label='-'
            value={'-'}
            action={pressCalcButton}
            backgroundColor='lightgrey'
          />
        </View>
        <View style={styles.row}>
          <CalcButton label='7' value={7} action={pressCalcButton} />
          <CalcButton label='8' value={8} action={pressCalcButton} />
          <CalcButton label='9' value={9} action={pressCalcButton} />
          <CalcButton
            label='x'
            value={'x'}
            action={pressCalcButton}
            backgroundColor='lightgrey'
          />
        </View>
        <View style={styles.row}>
          <CalcButton label='.' value={0.1} action={pressCalcButton} />
          <CalcButton label='0' value={0} action={pressCalcButton} />
          <CalcButton label='-/+' value={'-/+'} action={pressCalcButton} />
          <CalcButton
            label='/'
            value={'/'}
            action={pressCalcButton}
            backgroundColor='lightgrey'
          />
        </View>
        <View style={styles.row}>
          <CalcButton
            label='C'
            value={'C'}
            action={pressCalcButton}
            backgroundColor='orange'
          />
          <CalcButton
            label='AC'
            value={'AC'}
            action={pressCalcButton}
            backgroundColor={'firebrick'}
          />
          <CalcButton
            label='sqrt'
            value={'sqrt'}
            action={pressCalcButton}
            backgroundColor={'lightgrey'}
          />
          <CalcButton
            label='='
            value={'='}
            action={pressCalcButton}
            backgroundColor={'green'}
          />
        </View>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '65%',
    justifyContent: 'center',
    alignSelf: 'center',
    borderColor: 'black',
    borderWidth: 0.5,
  },
  row: {
    flexDirection: 'row',
    height: '20%',
    width: '25%',
  },
  column: {
    flexDirection: 'column',
    width: '100%',
    height: '100%',
    justifyContent: 'space-between',
    alignSelf: 'center',
  },
});
