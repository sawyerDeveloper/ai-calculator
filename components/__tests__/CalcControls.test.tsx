import { render } from '@testing-library/react-native';
import CalcControls from '../CalcControls';

it('renders correctly', () => {
  const tree = render(
    <CalcControls pressCalcButton={() => {}} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});