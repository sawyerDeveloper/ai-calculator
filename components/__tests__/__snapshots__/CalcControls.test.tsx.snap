// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "borderColor": "black",
      "borderWidth": 0.5,
      "height": "65%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "height": "20%",
          "width": "25%",
        }
      }
    >
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          1
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          2
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          3
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "lightgrey",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "height": "20%",
          "width": "25%",
        }
      }
    >
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          4
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          5
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          6
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "lightgrey",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "height": "20%",
          "width": "25%",
        }
      }
    >
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          7
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          8
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          9
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "lightgrey",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          x
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "height": "20%",
          "width": "25%",
        }
      }
    >
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          .
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 25,
              },
            ]
          }
        >
          0
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          -/+
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "lightgrey",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          /
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "height": "20%",
          "width": "25%",
        }
      }
    >
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "orange",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          C
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "firebrick",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          AC
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "lightgrey",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          sqrt
        </Text>
      </View>
      <View
        accessibilityLabel="Calculator Button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "black",
              "borderWidth": 1,
              "cursor": "pointer",
              "fontSize": 20,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
            {
              "backgroundColor": "green",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontWeight": "600",
              },
              {
                "fontSize": 30,
              },
            ]
          }
        >
          =
        </Text>
      </View>
    </View>
  </View>
</View>
`;
